Assets/_BForBoss/_Core/Scripts/Boss/DerekContextManager.cs

```diff
@@ -9,8 +9,8 @@ namespace BForBoss
     public class DerekContextManager : MonoBehaviour
```

Comment:
IMO, DerekContextManager cs DerekBossManager. If you ask someone what's the difference without explaining to them it's a bit confusing.

What if you called it DerekFiniteStateMachine?

Reply:
Yeah, so the way I have reasoned is the `ContextManager` is in charge of the current state of the encounter, Handle responding to the changes in the Labor/s (i.e. Ring labor) and Handle responding changes of the Boss.

So like if, for example, when the player completes a ring labor while Derek is phase 3 (the final phase) we want to change the lighting of the scene and play a different soundtrack, it would be the Context Manager who would be calling that. 

Thats why I am not so keen calling this class a FiniteStateMachine, even though that it what it basically is at the moment.

Reply:
I am down to change the name though! Here are some suggestions

`DerekDirector` (Director referring to the fact that this manager is basically in charge of all the main "Actors" in the scene and make them come together) - got this idea from a project I previously worked on
`DerekEnvirornmentManager` - A bit more on the nose


Down for other ideas

Reply:
Personal preference,i don't mind if u stick with context.